server.go

```diff
@@ -407,25 +417,9 @@ func (s *Server) runHandler(handler HandlerFunc, deliveries <-chan amqp.Delivery
 
 		ctx := context.WithValue(context.Background(), CtxQueueName, queueName)
 
-		// Use the default provided Acknowledger for the delivery
-		// (amqp.Channel) and add our ack aware acknowledger which can tell if
-		// a message has been acknowledged (ack, nack or rejected).
-		aac := ackAwareChannel{
```

Comment:
This is the only place where `ackAwareChannel` is used so if you want to remove it I guess we should remote the [whole file](https://github.com/bombsimon/amqp-rpc/blob/master/acknowledger.go) (and it's tests) as well. But should we? I think it's nice and if it's used you can type cast to an interface and check handled. If you add both our bundled middlewares now (panic recovery which will nack and auto ack which will ack) you'll get an error from the ack middleware. It's no biggie but it feels like bad design to ship with two middlewares that can't even co-exist.

Reply:
> If you add both our bundled middlewares now (panic recovery which will nack and auto ack which will ack) you'll get an error from the ack middleware.

Ahh, yes. But the `PanicRecovery` middleware won't work together with `AutoAck` anyway so I'll just remove the `Reject` and simplify it.

Reply:
👍 I guess this is why the `HandlerCrashedHeader` constant is there, so that users can look at the header and do ack/nack accordingly.